src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -106,7 +106,45 @@ public static void requireExactlyOneOf(Set<Modifier> modifiers, Modifier... mutu
         modifiers, Arrays.toString(mutuallyExclusive));
   }
 
-  public static boolean hasDefaultModifier(Collection<Modifier> modifiers) {
+  static boolean hasDefaultModifier(Collection<Modifier> modifiers) {
     return DEFAULT != null && modifiers.contains(DEFAULT);
   }
+
+  // {@linkplain https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.10.6}
+  static String characterLiteralWithoutSingleQuotes(char c) {
+    switch (c) {
+      case '\b': return "\\b"; /* \u0008: backspace (BS) */
+      case '\t': return "\\t"; /* \u0009: horizontal tab (HT) */
+      case '\n': return "\\n"; /* \u000a: linefeed (LF) */
+      case '\f': return "\\f"; /* \u000c: form feed (FF) */
+      case '\r': return "\\r"; /* \u000d: carriage return (CR) */
+      case '\"': return "\"";  /* \u0022: double quote (") */
+      case '\'': return "\\'"; /* \u0027: single quote (') */
+      case '\\': return "\\";  /* \u005c: backslash (\) */
+      default:
+      return Character.isISOControl(c) ? String.format("\\u%04x", (int) c) : Character.toString(c);
```

Comment:
I know ... but than it hits the 100 chars per line border. Will `import static Character.isISOControl()`.